src/api/index.js

```diff
@@ -0,0 +1,142 @@
+import { echo } from '@theholocron/klaxon';
+import render from './render';
+import {
+	AuctionConf,
+	InterestGroup,
+} from './types';
+import {
+	frame,
+	message,
+	validate,
+} from './utils';
+
+/*
+* @const {number}
+* @summary Maximum expiration allowed for an Interest Group to exist
+* @description Milliseconds occuring per day multiplied by the maximum number of days (maximum dayus (30) * hours per day (24) * minutes per hour (60) * seconds per minute (60) * milliseconds per second (1000))
+*/
+const MAX_EXPIRATION = 2592000000;
+
+/*
+* @const {URL}
+* @description The source URL for the hosted iframe
+*/
+const IFRAME_HOST = 'http://localhost:3000/docs/iframe.html';
+
+export default class Fledge {
+	constructor (debug) {
+		const query = debug ? '?debug=true' : '';
+		const { iframe, origin } = frame.create({
+			source: `${IFRAME_HOST}${query}`,
+			style: { display: 'none' },
+		});
+		// iframe.sandbox.add('allow-same-origin', 'allow-scripts');
+		const port = message.getFramePort(iframe, origin);
+
+		this._debug = debug;
+		this._props = {
+			iframe,
+			port,
+		};
+	}
+
+	set props (props) {
+		this._props = props;
+	}
+
+	get props () {
+		return this._props;
+	}
+
+	async joinAdInterestGroup (options, expiry) {
```

Comment:
what happened to all the great doc blocks you had? the external facing code needs them more than anything.